### PR TITLE
Allow puppetserver to be used with foss P3

### PR DESCRIPTION
**Pull request by hunner**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ group :test do
   gem "puppetlabs_spec_helper"
   gem "metadata-json-lint"
   gem "rspec-puppet-facts"
-  gem 'rubocop', '0.33.0'
```
Comment (jyaworski) [inline Gemfile]: You want to pin this. Trust me. To some version.

File: manifests/install.pp [hunk 2]
```diff
@@ -2,71 +2,71 @@
...
+      if $gem_source {
+        # Use a local source, like the package providers would
+        validate_absolute_path($gem_source)
+        $source_flag = '--local'
+      } else {
+        $source_flag = undef
+      }
+      exec { "install puppetserver gem ${gem_name}":
+        command => "puppetserver gem install ${source_flag} ${gem_name} ${gem_flag}",
+        unless  => "find ${puppetserver_gem_dir} -maxdepth 1 -type d -regextype posix-egrep -regex '.*/${name}-([0-9].?)+'"
```
Comment (jyaworski) [inline manifests/install.pp]:
```shell
[root@puppet1 gems]# find /var/lib/puppet/jruby-gems/gems/ -maxdepth 1 -type d -regextype posix-egrep -regex ".*/hiera-eyaml-([[:digit:]].?)+"
/var/lib/puppet/jruby-gems/gems/hiera-eyaml-2.1.0
```

**Comment by jyaworski** [comment]
It looks like lint is failing.

File: Gemfile (hunk 1 above)
Comment (alexjfisher) [inline Gemfile]: @jyaworski - I think this is why the build is failing for me in https://github.com/hunner/puppet-hiera/pull/127

**Pull request merged**